network/src/application/storage.rs

```diff
@@ -2,166 +2,199 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use crate::{
-    application::types::{PeerInfo, PeerState},
-    transport::ConnectionMetadata,
-};
-use aptos_config::{
-    config::Error,
-    network_id::{NetworkId, PeerNetworkId},
+    application::{
+        error::Error,
+        metadata::{ConnectionState, PeerMetadata},
+    },
+    transport::{ConnectionId, ConnectionMetadata},
+    ProtocolId,
 };
+use aptos_config::network_id::{NetworkId, PeerNetworkId};
 use aptos_infallible::RwLock;
-use aptos_types::{account_address::AccountAddress, PeerId};
+use aptos_types::PeerId;
 use std::{
     collections::{hash_map::Entry, HashMap},
-    fmt::Debug,
     sync::Arc,
 };
 
-// TODO: refactor and clean up this interface.
-
-/// Metadata storage for peers across all of networking.  Splits storage of information across
-/// networks to prevent different networks from affecting each other
+/// A simple container that tracks all peers and peer metadata for the node.
+/// This container is updated by both the networking code (e.g., for new
+/// peer connections and lost peer connections), as well as individual
+/// applications (e.g., peer monitoring service).
 #[derive(Debug)]
-pub struct PeerMetadataStorage {
-    storage: HashMap<NetworkId, RwLock<HashMap<PeerId, PeerInfo>>>,
+pub struct PeersAndMetadata {
```

Comment:
nit pick: the name says nothing about the NetworkId. Should it be `NetworkPeersAndMetadata`?

Reply:
Interesting... I think `PeersAndMetadata` is okay, but yeah, maybe `NetworkPeersAndMetadata` is a little more obvious. I'll send out a PR once I flush the stack 😄